src/fix/main.rs

```diff
@@ -0,0 +1,1059 @@
+
+#[derive(Debug, Clone)]
+pub struct CLIOptions {
+    game_id: Option<String>,
+    japanese: bool,
+    new_licensee: Option<String>,
+    old_licensee: Option<u8>,
+    cartridge_type: MbcType,
+    rom_version: Option<u8>,
+    overwrite_rom: bool,
+    pad_value: Option<u8>,
+    ram_size: Option<u8>,
+    sgb: bool,
+    fix_spec: Vec<FixSpec>,
+    model: Model,
+    title: Option<String>,
+    title_len: u8,
+}
+
+#[derive(Parser, Debug)]
+#[clap(version = "1.0", help = "A tool to fix ROM files for Game Boy.")]
+struct Cli {
+    #[clap(short = 'C', long = "color-only", help = "Color-only mode")]
+    color_only: bool,
+
+    #[clap(short = 'c', long = "color-compatible", help = "Color-compatible mode")]
+    color_compatible: bool,
+
+    #[clap(short = 'f', long = "fix-spec", help = "Specify a fix specification", value_name = "FIX_SPEC")]
+    fix_spec: Option<String>,
+
+    #[clap(short = 'i', long = "game-id", help = "Specify a game ID", value_name = "GAME_ID")]
+    game_id: Option<String>,
+
+    #[clap(short = 'j', long = "non-japanese", help = "Non-Japanese mode")]
+    non_japanese: bool,
+
+    #[clap(short = 'k', long = "new-licensee", help = "Specify a new licensee", value_name = "NEW_LICENSEE")]
+    new_licensee: Option<String>,
+
+    #[clap(short = 'l', long = "old-licensee", help = "Specify an old licensee", value_name = "OLD_LICENSEE", value_parser=maybe_hex::<u8>)]
+    old_licensee: Option<u8>,
+
+    #[clap(short = 'm', long = "mbc-type", help = "Specify the MBC type", value_name = "MBC_TYPE")]
+    mbc_type: Option<String>, //TOCHECK: is parsing the string required or does clap do it for me?
+
+    #[clap(short = 'n', long = "rom-version", help = "Specify the ROM version", value_name = "ROM_VERSION", value_parser=maybe_hex::<u8>)]
+    rom_version: Option<u8>,
+
+    #[clap(short = 'O', long = "overwrite", help = "Overwrite the file")]
+    overwrite: bool,
+
+    #[clap(short = 'p', long = "pad-value", help = "Specify the padding value", value_name = "PAD_VALUE", value_parser=maybe_hex::<u8>)]
+    pad_value: Option<u8>,
+
+    #[clap(short = 'r', long = "ram-size", help = "Specify the RAM size", value_name = "RAM_SIZE", value_parser=maybe_hex::<u8>)]
+    ram_size: Option<u8>,
+
+    #[clap(short = 's', long = "sgb-compatible", help = "SGB-compatible mode")]
+    sgb_compatible: bool,
+
+    #[clap(short = 't', long = "title", help = "Specify the title", value_name = "TITLE")]
+    title: Option<String>,
+
+    #[clap(short = 'V', long = "version", help = "Print RGBFIX version and exit")]
+    version: bool,
+
+    #[clap(short = 'v', long = "validate", help = "Fix the header logo and both checksums (-f lhg)")]
+    validate: bool,
+
+    #[clap(help = "Files to process")]
+    files: Vec<String>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum FixSpec {
+    FixLogo,
+    TrashLogo,
+    FixHeaderSum,
+    TrashHeaderSum,
+    FixGlobalSum,
+    TrashGlobalSum,
+}
+
+macro_rules! logo {
```

Comment:
This macro hurts (my fault for writing it); I wanna find a better way to do this even if it's at runtime.